app/src/androidTest/java/com/example/househomey/DateFilterFragmentTest.java

```diff
@@ -0,0 +1,114 @@
+package com.example.househomey;
+
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static com.example.househomey.testUtils.TestHelpers.hasListLength;
+import static com.example.househomey.testUtils.TestHelpers.waitFor;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Before;
+import org.junit.Test;
+import com.example.househomey.testUtils.TestSetup;
+
+public class DateFilterFragmentTest extends TestSetup  {
+    @Before
+    public void waitForItems() {
+        waitFor(() -> {
+            onView(withId(R.id.item_list)).check(matches(isDisplayed()));
+        });
+        navigateToDateRangeFilter();
+    }
+
+    @Test
+    public void testDateFilterWithResults() {
+        setDateFilter(R.id.start_date_filter, "10/01/2023");
+        setDateFilter(R.id.end_date_filter, "10/30/2023");
+        onView(withText("APPLY")).perform(click());
+        waitFor(() -> hasListLength(3));
+    }
+
+    @Test
+    public void testDateFilterWithNoResults() {
+        setDateFilter(R.id.start_date_filter, "10/01/2001");
+        setDateFilter(R.id.end_date_filter, "10/01/2001");
+        onView(withText("APPLY")).perform(click());
+        waitFor(() -> hasListLength(0));
+    }
+
+    @Test
+    public void testDateFilterWithOneDateFilled() {
+        setDateFilter(R.id.start_date_filter, "10/01/2023");
+        onView(withText("APPLY")).perform(click());
+        waitFor(() -> hasListLength(4));
```

Comment:
hasListLength is actually a bit problematic of a test. It only checks how many items are _visible_, not how many items are in the list. It would be better for hasListLength to check the list count element and verify it is correct in everything except the list count espresso test.

If we just change hasListLength to that we should be good.

I also believe this is why your test fails CI

Reply:
Tx for pointing that out. I made the change to `hasListLength`